src/Resources/ResourceManager/Implementation/ResourceGroups/GetAzureResourceGroupCmdlet.cs

```diff
@@ -65,8 +65,13 @@ public class GetAzureResourceGroupCmdlet : ResourceManagerCmdletBaseWithApiVersi
 
         protected override void OnProcessRecord()
         {
-            Name = Name ?? ResourceIdentifier.FromResourceGroupIdentifier(this.Id).ResourceGroupName;
-            this.WriteObject(ResourceManagerSdkClient.FilterResourceGroups(name: this.Name, tag: this.Tag, detailed: false, location: this.Location), true);
+            String subscriptionId = null;
+            if (String.IsNullOrEmpty(Name) && !String.IsNullOrEmpty(this.Id)){
```

Comment:
Use something like `if (ParameterSetName == "ById")` instead because comparing parameter set name is more clear.